### DISCUSSION TITLE
How to trigger toast in server component?

### DISCUSSION THREAD

**Discussion by zeekrey** [shadcn-ui/ui#2136] [Q&A]
Hey there! 😊

I'm wondering what's the most effective way to display a toast message based on data known in a server component.

-  It seems like calling `toast()` might not be feasible in a server component, as `Toaster` is specifically a client component, even if `rsc` is set to true.
-  Running `toast()` within `useEffect` doesn't seem like the best option either, since I want to avoid displaying the toast every time useEffect runs. I'd prefer to batch the toasts and show them only once.

**Comment by jatinder-singh221** [comment]
> Hey there! 😊
> 
> I'm wondering what's the most effective way to display a toast message based on data known in a server component.
> 
> * It seems like calling `toast()` might not be feasible in a server component, as `Toaster` is specifically a client component, even if `rsc` is set to true.
> * Running `toast()` within `useEffect` doesn't seem like the best option either, since I want to avoid displaying the toast every time useEffect runs. I'd prefer to batch the toasts and show them only once.

you can use a custom hook like this

`export default function useFormEvent(action: ActionType) {
    const [state, FormAction] = useFormState(action, null)

// do anything with your state
return {FormAction}
}
`
